src/model/id.rs

```diff
@@ -112,10 +123,13 @@ macro_rules! id_u64 {
             }
 
             impl std::str::FromStr for $name {
-                type Err = <u64 as std::str::FromStr>::Err;
+                type Err = IDFromStrError;
 
                 fn from_str(s: &str) -> Result<Self, Self::Err> {
-                    Ok(Self(s.parse()?))
+                    #[cfg(debug_assertions)]
+                    { return Ok(Self(s.parse::<u64>().ok().and_then(NonZeroU64::new).ok_or(IDFromStrError)?)) }
+                    #[cfg(not(debug_assertions))]
+                    { return Ok(Self(snowflake::parse(s).ok_or(IDFromStrError)?)) }
```

Comment:
I dislike this, we should always rely on the dangerous path when there are debug assertions as otherwise that path will hide bugs.

Reply:
I, uh, cant.

Reply:
What do you mean? You can't remove the `#[cfg]`?

Reply:
It would break this test

https://github.com/serenity-rs/serenity/blob/c5e6b3d923043e0028b8ef3c2315becbafd0670d/src/utils/mod.rs#L130-L131

Reply:
You have a good point.